In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('../data/StudentPerformanceFactors.csv')

In [3]:
df.head()

df.info()

df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6607 entries, 0 to 6606
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Hours_Studied               6607 non-null   int64 
 1   Attendance                  6607 non-null   int64 
 2   Parental_Involvement        6607 non-null   object
 3   Access_to_Resources         6607 non-null   object
 4   Extracurricular_Activities  6607 non-null   object
 5   Sleep_Hours                 6607 non-null   int64 
 6   Previous_Scores             6607 non-null   int64 
 7   Motivation_Level            6607 non-null   object
 8   Internet_Access             6607 non-null   object
 9   Tutoring_Sessions           6607 non-null   int64 
 10  Family_Income               6607 non-null   object
 11  Teacher_Quality             6529 non-null   object
 12  School_Type                 6607 non-null   object
 13  Peer_Influence              6607 non-null   obje

,Hours_Studied,Attendance,Sleep_Hours,Previous_Scores,Tutoring_Sessions,Physical_Activity,Exam_Score
count,6607.000000,6607.000000,6607.00000,6607.000000,6607.000000,6607.000000,6607.000000
mean,19.975329,79.977448,7.02906,75.070531,1.493719,2.967610,67.235659
std,5.990594,11.547475,1.46812,14.399784,1.230570,1.031231,3.890456
min,1.000000,60.000000,4.00000,50.000000,0.000000,0.000000,55.000000
25%,16.000000,70.000000,6.00000,63.000000,1.000000,2.000000,65.000000
50%,20.000000,80.000000,7.00000,75.000000,1.000000,3.000000,67.000000
75%,24.000000,90.000000,8.00000,88.000000,2.000000,4.000000,69.000000
max,44.000000,100.000000,10.00000,100.000000,8.000000,6.000000,101.000000


## Data Overview

The dataset contains 6,607 student records with 20 features including demographic, academic, and environmental factors. The average exam score is 67.24 with a standard deviation of 3.89. Scores range from 55 to 101. There are some missing values in Teacher_Quality, Parental_Education_Level, and Distance_from_Home.

In [4]:
fig = px.histogram(df, x='Exam_Score', nbins=20, title='Distribution of Exam Scores')
fig.show()

## Distribution of Exam Scores

The exam scores are approximately normally distributed, with most students scoring between 65 and 69. The median score is 67.

In [5]:
fig = px.bar(df['Gender'].value_counts(), title='Gender Distribution')
fig.show()

## Gender Distribution

The dataset includes 3,814 male and 2,793 female students, with a slight majority of males.

In [6]:
fig = px.box(df, x='Gender', y='Exam_Score', title='Exam Scores by Gender')
fig.show()

## Exam Scores by Gender

Male and female students show similar distributions in exam scores, with comparable medians and ranges.

In [7]:
fig = px.box(df, x='Parental_Education_Level', y='Exam_Score', title='Exam Scores by Parental Education Level')
fig.show()

## Exam Scores by Parental Education Level

Students whose parents have postgraduate education show slightly higher median exam scores compared to those with high school or college-educated parents.

In [8]:
numeric_cols = df.select_dtypes(include=[np.number]).columns
corr_matrix = df[numeric_cols].corr()
fig = px.imshow(corr_matrix, text_auto=True, title='Correlation Matrix')
fig.show()

## Correlation Analysis

Attendance has the strongest positive correlation with exam scores (0.58), followed by hours studied (0.45). Previous scores and tutoring sessions show moderate positive correlations, while sleep hours and physical activity have weak correlations.